core/2d/ActionManager.h

```diff
@@ -36,9 +36,14 @@ THE SOFTWARE.
 
 NS_AX_BEGIN
 
-class Action;
-
-struct _hashElement;
+struct _ActionHandle
```

Comment:
Identifiers beginning with an underscore followed by an uppercase letter are always reserved in C++. You should drop the underscore here. If the idea is to have some "internal" type I would suggest to use a `detail` namespace.